CleverPush/Source/CleverPushInstance.m

```diff
@@ -1514,24 +1515,50 @@ - (void)handleNotificationOpened:(NSDictionary*)payload isActive:(BOOL)isActive
             [CPAppBannerModuleInstance setCurrentVoucherCodePlaceholder:voucherCodesByAppBanner];
         }
 
-        [self showAppBanner:[notification valueForKey:@"appBanner"] channelId:[payload cleverPushStringForKeyPath:@"channel._id"] notificationId:notificationId];
-    }
-
-    CPNotificationOpenedResult * result = [[CPNotificationOpenedResult alloc] initWithPayload:payload action:action];
-
-    if (!channelId) { // not init
-        pendingOpenedResult = result;
+        [self showAppBanner:[notification valueForKey:@"appBanner"] channelId:[payloadMutable cleverPushStringForKeyPath:@"channel._id"] notificationId:notificationId];
     }
 
     if (notification != nil && [notification objectForKey:@"url"] != nil && ![[notification objectForKey:@"url"] isKindOfClass:[NSNull class]] && [[notification objectForKey:@"url"] length] != 0) {
         NSURL *url = [NSURL URLWithString:[notification objectForKey:@"url"]];
         if ([notification objectForKey:@"autoHandleDeepLink"] != nil && ![[notification objectForKey:@"autoHandleDeepLink"] isKindOfClass:[NSNull class]] && [[notification objectForKey:@"autoHandleDeepLink"] boolValue]) {
-            if ([[UIApplication sharedApplication] canOpenURL:url]) {
-                [[UIApplication sharedApplication] openURL:url options:@{} completionHandler:nil];
-            } 
+
+            if (action != nil && ![action isKindOfClass:[NSNull class]] && [notification objectForKey:@"actions"] != nil && ![[notification objectForKey:@"actions"] isKindOfClass:[NSNull class]] && [[notification objectForKey:@"actions"] isKindOfClass:[NSArray class]] && [[notification objectForKey:@"actions"] count] > 0) {
```

Comment:
- Put into extra function
- Make if condition more clear, also in extra function